registration-processor/pre-processor/registration-processor-packet-validator-stage/src/main/java/io/mosip/registration/processor/stages/utils/AuditUtility.java

```diff
@@ -0,0 +1,117 @@
+package io.mosip.registration.processor.stages.utils;
+
+import java.io.InputStream;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.function.Supplier;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.type.CollectionType;
+
+import io.mosip.kernel.core.logger.spi.Logger;
+import io.mosip.registration.processor.core.constant.LoggerFileConstant;
+import io.mosip.registration.processor.core.constant.PacketFiles;
+import io.mosip.registration.processor.core.logger.RegProcessorLogger;
+import io.mosip.registration.processor.core.packet.dto.AuditDTO;
+import io.mosip.registration.processor.core.packet.dto.AuditRequestDTO;
+import io.mosip.registration.processor.core.packet.dto.AuditRespDTO;
+import io.mosip.registration.processor.core.spi.filesystem.manager.PacketManager;
+import io.mosip.registration.processor.stages.dto.RestRequestDTO;
+import io.mosip.registration.processor.stages.helper.RestHelper;
+import io.mosip.registration.processor.stages.packet.validator.PacketValidateProcessor;
+
+/**
+ * @author Tapaswini Behera M1043226
+ *
+ */
+
+@Component
+public class AuditUtility {
+
+	/** The reg proc logger. */
+	private static Logger regProcLogger = RegProcessorLogger.getLogger(PacketValidateProcessor.class);
+
+	@Autowired
+	private RestHelper restHelper;
+
+	@Autowired
+	private Environment env;
+
+	@Autowired
+	private PacketManager fileSystemManager;
+	
+	@Autowired
+	private ObjectMapper mapper;
+	
+	@Async
+	public void saveAuditDetails(String registrationId) {
+		try {
+			regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(),
+					"", "AuditUtility::saveAuditDetails()::entry");
+			
+			
+			InputStream auditFileInputStream = fileSystemManager.getFile(registrationId, PacketFiles.AUDIT.name());
+			CollectionType collectionType = mapper.getTypeFactory().constructCollectionType(List.class,
+					AuditDTO.class);
+			
+			List<AuditDTO> regClientAuditDTOs = mapper.readValue(auditFileInputStream, collectionType);
+			regClientAuditDTOs.parallelStream().forEach(audit -> {
+				RestRequestDTO request = buildRequest(audit);
+				Supplier<Object> dto = restHelper.requestAsync(request);
+				dto.get();
+				
+			});
+			System.out.println("--------------------------save in db-----");
+		} catch (Exception e) {
+			System.out.println(ExceptionUtils.getStackTrace(e));
+			regProcLogger.error(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(),
```

Comment:
add audit id in exception log so that in case of exception its easy to identify which audits failed.